packages/elements/test/integration/realtime-reporting.it.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { expect } from 'chai';
+import { SseTestServer } from './lib/SseServer';
+import { getCurrent } from './lib/browser';
+import { ReportPage } from './po/ReportPage';
+import { ReportingClient } from './lib/SseServer';
+import { sleep } from './lib/helpers';
+import { MutantStatus } from 'mutation-testing-report-schema';
+
+describe('realtime reporting', () => {
+  const server: SseTestServer = new SseTestServer();
+  const defaultEvent = { id: '0', status: 'Killed' };
+
+  let port: number;
+  let page: ReportPage;
+  let client: ReportingClient;
+
+  beforeEach(async () => {
+    port = await server.start();
+    page = new ReportPage(getCurrent());
+  });
+
+  describe('when navigating to the overview page', () => {
+    it('should update the mutation testing metrics', async () => {
+      server.on('client-connected', (c) => (client = c));
+
+      await page.navigateTo(`realtime-reporting-example/?port=${port}`);
+      await page.whenFileReportLoaded();
+      client.sendMutantTested(defaultEvent);
+      client.sendMutantTested({ id: '1', status: 'Survived' });
+
+      const allFilesRow = page.mutantView.resultTable().row('All files');
+      const attributesRow = page.mutantView.resultTable().row('Attributes');
+      const wrappitContextRow = page.mutantView.resultTable().row('WrappitContext.cs');
+
+      expect(await allFilesRow.mutationScore()).to.eq('50.00');
+      expect(await attributesRow.mutationScore()).to.eq('100.00');
+      expect(await wrappitContextRow.mutationScore()).to.eq('0.00');
+    });
+  });
+
+  describe('when navigating to a file with 1 mutant', () => {
+    it('should update the state of a mutant', async () => {
+      server.on('client-connected', (c) => (client = c));
+      await page.navigateTo(`realtime-reporting-example/?port=${port}#mutant/Attributes/HandleAttribute.cs/`);
+      await page.whenFileReportLoaded();
+      expect((await page.mutantView.mutantDots()).length).to.equal(1);
+      const mutantPending = page.mutantView.mutantMarker('0');
+      expect(await mutantPending.underlineIsVisible()).to.be.true;
+
+      client.sendMutantTested(defaultEvent);
+      await sleep(25);
```

Comment:
Can we refactor so we don't need a magic sleep?

Reply:
Hmm I'm thinking of a way to do this, but you have to do this since sending an event to the browser takes time (and the call is not blocking).. 🤔